consensus/vbft/chain_store.go

```diff
@@ -48,13 +72,50 @@ func (self *ChainStore) GetChainedBlockNum() uint32 {
 	return self.chainedBlockNum
 }
 
+func (self *ChainStore) SetExecMerkleRoot(blkNum uint32, merkleRoot common.Uint256) {
+	if blk, present := self.pendingBlocks[blkNum]; blk != nil && present {
+		blk.execResult.MerkleRoot = merkleRoot
+	} else {
+		log.Errorf("SetExecMerkleRoot failed blkNum:%d,merkleRoot:%s", blkNum, merkleRoot.ToHexString())
+	}
+}
+
+func (self *ChainStore) GetExecMerkleRoot(blkNum uint32) common.Uint256 {
+	if blk, present := self.pendingBlocks[blkNum]; blk != nil && present {
+		return blk.execResult.MerkleRoot
+	}
+	merkleRoot, err := self.server.ledger.GetStateMerkleRoot(blkNum)
+	if err != nil {
+		log.Errorf("GetStateMerkleRoot blockNum:%d, error :%s", blkNum, err)
+		return common.Uint256{}
+	} else {
+		return merkleRoot
+	}
+
+}
+
+func (self *ChainStore) SetExecWriteSet(blkNum uint32, memdb *overlaydb.MemDB) {
+	if blk, present := self.pendingBlocks[blkNum]; blk != nil && present {
```

Comment:
suggest to change to setExecWriteSet()

Reply:
If it is only used by OpenBlockStore(), suggest to remove the function, because this function update the PendingBlock **partially**.

Reply:
has modify